src/modules/exec.ts

```diff
@@ -126,7 +126,10 @@ export async function runExecSpec(
     }
 
     try {
-        const command = programSpec.executable.includes('/') ? resolve(cwd, programSpec.executable) : programSpec.executable;
+        let command = programSpec.executable.includes('/') ? resolve(cwd, programSpec.executable) : programSpec.executable;
+
+        command = await ensureVenvIfPython(command, envVars, cwd, loggingOptions);
```

Comment:
The name `ensureVEnvIfPython` suggests that something is *ensured* whereas there is actually a lot of side effects going on when calling the function. For a better separation of concerns and better testability, why don't we introduce:

`isPython(executable: string): bool`
`createVenv(...): string`

and call it like this

```TypeScript
let command = programSpec.executable.....
if (isPython(command.executable)) {
  command = createVenv(...)
}
const result = await awaitSpawn(command, ...);
```

Reply:
(My intention was to have just one additional line in the "original" `runExecSpec`